packages/mdl-ripple/index.js

```diff
@@ -39,67 +21,123 @@ export default class MDLRipple {
     background.classList.add(Class.BACKGROUND);
     root.appendChild(background);
 
-    this.elements_ = {
-      [Identifier.ROOT]: root,
-      [Identifier.BACKGROUND]: background,
-      [Identifier.SURFACE]: surface,
-    };
-
     for (let i = 0; i < MAX_RIPPLES; i++) {
       let foreground = document.createElement('div');
       foreground.classList.add(Class.FOREGROUND);
       let foregroundCircle = document.createElement('div');
       foregroundCircle.classList.add(Class.FOREGROUND_CIRCLE);
       foreground.appendChild(foregroundCircle);
       root.appendChild(foreground);
+    }
+    return root;
+  }
 
-      this.elements_[this.get(Identifier.FOREGROUND, i)] = foreground;
-      this.elements_[this.get(Identifier.FOREGROUND_CIRCLE, i)] = foregroundCircle;
+
+  /**
+   * @param {!Element} surface
+   * @param {!Element} root
+   * @return {!MDLRipple}
+   */
+  static attachTo(surface, root) {
+    let elements = {
+      [Identifier.SURFACE]: surface,
+      [Identifier.ROOT]: root,
+      [Identifier.BACKGROUND]: root.getElementsByClassName(Class.BACKGROUND)[0]
+    };
+
+    let foregrounds = root.getElementsByClassName(Class.FOREGROUND);
+    let foregroundCircles = root.getElementsByClassName(Class.FOREGROUND_CIRCLE);
+    for (let i = 0; i < foregrounds.length; i++) {
+      elements[ref(Identifier.FOREGROUND, i)] = foregrounds[i];
+      elements[ref(Identifier.FOREGROUND_CIRCLE, i)] = foregroundCircles[i];
     }
+
+    return new MDLRipple(elements);
   }
 
+
+  /** @param {!Object} elements */
+  constructor(elements) {
+    super(elements[Identifier.ROOT]);
+
+    this.elements_ = elements;
+
+    this.initMdlRipple_();
+
+    let surface = elements[Identifier.SURFACE];
+
+    this.isBounded = surface.getAttribute('bounded') != 'false';
```

Comment:
hmm I could foresee cases where users may want to pass this in programmatically. Perhaps it would be better to pass this into the constructor as an options object, and then have a method like `attachTo()` do the attribute checking?